### PR TITLE
tests: fix docu and coding style

### PR DESCRIPTION
**Pull request by smlng**
- update documentation of all tests
- enforce kernel coding style on all tests
- add all tests to coding style check

### CODE REVIEW

**Comment by smlng** [comment]
This PR fixes issues #3, #48, and #63 for all tests.

**Comment by waehlisch** [comment]
Thanks @smlng. Can you be a bit more specific in the documentation? For example, "Compare SPKI records" is very short (do you compare equal values, different values etc.?).

**Comment by smlng** [comment]
Update and extend documentation of tests, please review.

**Comment by waehlisch** [comment]
I'm still not very happy ;). For example, 

```
/*
+ * @brief lpfst_test
+ * Run tests to validate a longest prefix first search tree
+ */
```

The additional comment does not give many more insights. What is the test about (instead of testing the lpfst, which is already encoded in the function name)?

Maybe it is a general question: For me the comments should explain the test briefly but in sufficient details such that someone understands what happen without reading the code. What is the input? What is the expected output? However, you can also argue that the code is self-explanatory.

What do we want? @fho @nbars comments?

**Comment by smlng** [comment]
@waehlisch, I elaborated a bit more on the test docu. While it's good to have a short/rough description what each tests, I don't think we want to run doxygen on the test code, or do we?!

**Comment by waehlisch** [comment]
@fho @nbars what do you prefer?

**Comment by fho** [comment]
I agree with @smlng. It's unlikely that someone would read the Doxygen documentation for the tests.
Best would be if the function signature is self-explanatory and we could go without any extra descriptions

**Comment by smlng** [comment]
@fho: please clarify: are the extended docu/comments of tests in this PR sufficient? If so, we should be clear to merge?!

File: tests/test_ht_spkitable.c [hunk 1]
```diff
@@ -1,391 +1,450 @@
...
+static struct spki_record *create_record(int ASN,
+					 int ski_offset,
+					 int spki_offset,
+					 struct rtr_socket *socket);
+/**
+ * @brief Compare SPKI records for equality
+ *
+ * @return true if r1 == r2, false otherwise
+  */
+static bool compare_spki_records(struct spki_record *r1,
```
Comment (fho) [inline tests/test_ht_spkitable.c]: If the function is renamed to bool spki_records_are_equal(), the function signature would contain the same information than the current description, therefore the description can be omitted.

File: tests/test_ht_spkitable.c [hunk 2]
```diff
@@ -1,391 +1,450 @@
...
+		return false;
+	if (memcmp(r1->spki, r2->spki, SPKI_SIZE) != 0)
+		return false;
+
+	return true;
 }
 
-static struct spki_record *create_record(int ASN, int ski_offset, int spki_offset, struct rtr_socket *socket)
+/**
+ * @brief Create a SPKI record
```
Comment (fho) [inline tests/test_ht_spkitable.c]: The description doesn't provide any additional value compared to the informations you get out of the function signature.

**Comment by fho** [comment]
I created a few comments in the commit few.

Documentation is always hard to maintain and becomes quickly outdated.
Therefore we should only provide comments if we are not able to make the code self-explanatory.
Having documentation that doesn't add informatinal value is only a burden, makes the project harder to maintain.

File: tests/test_ht_spkitable.c [hunk 3]
```diff
@@ -1,391 +1,450 @@
...
+ *
+ * @return new SPKI record
```
Comment (fho) [inline tests/test_ht_spkitable.c]: Adds no additionals information, same is provided by the function signature

File: tests/test_ht_spkitable.c [hunk 4]
```diff
@@ -1,391 +1,450 @@
...
+	return record;
 }
 
 /**
- * @brief test_ht_1
- * Test if the spki_table_src_remove function is working correctly by adding
- * spki_record associated with different rtr_socket. Then call spki_table_src_remove
- * with one of the sockets as argument and validate that all records associated with different
- * sockets are still there.
+ * @brief Helper shortcut, to assert insert of table entry
```
Comment (fho) [inline tests/test_ht_spkitable.c]: Could be helpful to description what does it assert exacty, when will the assert fail?

File: tests/test_ht_spkitable.c [hunk 5]
```diff
@@ -1,391 +1,450 @@
...
-    spki_table_free(&table);
-    free(result);
-    free(socket_one);
-    free(socket_two);
+/* ----- TESTS FUNCTIONS ----- */
 
-    printf("test_h1_1() complete\n");
+/**
+ * @brief test_ht_1
+ * Test if the spki_table_src_remove function is working correctly by adding
```
Comment (fho) [inline tests/test_ht_spkitable.c]: helpful description, makes it easier to find out what the function is doing

File: tests/test_pfx.c [hunk 6]
```diff
@@ -30,324 +18,404 @@
...
+	}
+
+	/* cleanup: free table */
+	pfx_table_free(&pfxt);
+	printf("mass_test successful\n");
 }
 
-int main()
+/**
+ * @brief Test basic ops on prefix table
```
Comment (fho) [inline tests/test_pfx.c]: Which ops are basic?

Comment (fho) [inline tests/test_pfx.c]: the brief line can be replaced with the following line.
The first line doesn't provide any information

**Comment by smlng** [comment]
Thanks @fho for reviewing. I included most of your suggestion and updated the PR. 

Even though some comments are redundant with signature of the function they describe; I would rather opt for consistency in documentation, i.e., at least a brief description for each function.

**Comment by smlng** [comment]
No further comments for about a week; @fho, @waehlisch: any objections to merging this one?

**Comment by fho** [comment]
I'm fine with it.
In general it would be good to make smaller commits for each logicial change and explain the commit message why it was done:
e.g.:

- commit for reworking the documentation (commit message could contain why it was decided to also add small descriptions for functions that don't contain additional informations)
- commit for adapting the coding style (should even done per module (tests, lib, pfx, scripts, etc)

It makes it easier to review changes, to understand the reason behind it and allows later to revert  specific changes if needed.

**Pull request merged**